resources/util.ts

```diff
@@ -0,0 +1,152 @@
+    DRG: 22,
+    BRD: 23,
+    WHM: 24,
+    BLM: 25,
+    ACN: 26,
+    SMN: 27,
+    SCH: 28,
+    ROG: 29,
+    NIN: 30,
+    MCH: 31,
+    DRK: 32,
+    AST: 33,
+    SAM: 34,
+    RDM: 35,
+    BLU: 36,
+    GNB: 37,
+    DNC: 38,
+  };
+
+  const tankJobs: Job[] = ['GLA', 'PLD', 'MRD', 'WAR', 'DRK', 'GNB'];
+  const healerJobs: Job[] = ['CNJ', 'WHM', 'SCH', 'AST'];
+  const meleeDpsJobs: Job[] = ['PGL', 'MNK', 'LNC', 'DRG', 'ROG', 'NIN', 'SAM'];
+  const rangedDpsJobs: Job[] = ['ARC', 'BRD', 'DNC', 'MCH'];
+  const casterDpsJobs: Job[] = ['BLU', 'RDM', 'BLM', 'SMN', 'ACN', 'THM'];
+  const dpsJobs: Job[] = [...meleeDpsJobs, ...rangedDpsJobs, ...casterDpsJobs];
+  const craftingJobs: Job[] = ['CRP', 'BSM', 'ARM', 'GSM', 'LTW', 'WVR', 'ALC', 'CUL'];
+  const gatheringJobs: Job[] = ['MIN', 'BTN', 'FSH'];
+
+  const stunJobs: Job[] = ['BLU', ...tankJobs, ...meleeDpsJobs];
+  const silenceJobs: Job[] = ['BLU', ...tankJobs, ...rangedDpsJobs];
+  const sleepJobs: Job[] = ['BLM', 'BLU', ...healerJobs];
+  const feintJobs: Job[] = [...meleeDpsJobs];
+  const addleJobs: Job[] = [...casterDpsJobs];
+  const cleanseJobs: Job[] = ['BLU', 'BRD', ...healerJobs];
+
+  const jobToRoleMap: Map<Job, Role> = (() => {
+    const addToMap = (map: Map<Job, Role>, jobs: Job[], role: Role) => {
+      jobs.forEach((job) => map.set(job, role));
+    };
+
+    const map: Map<Job, Role> = new Map([['NONE', 'none']]);
+    addToMap(map, tankJobs, 'tank');
+    addToMap(map, healerJobs, 'healer');
+    addToMap(map, dpsJobs, 'dps');
+    addToMap(map, craftingJobs, 'crafter');
+    addToMap(map, gatheringJobs, 'gatherer');
+
+    return map;
+  })();
+
+  return {
+    jobEnumToJob: (id: number) => allJobs.find((job: Job) => nameToJobEnum[job] === id),
+    jobToJobEnum: (job: Job): number => nameToJobEnum[job],
+    jobToRole: (job: Job): Role => {
+      const role = jobToRoleMap.get(job);
+      if (role === undefined)
+        return 'none';
+      return role;
+    },
+    getAllRoles: (): readonly Role[] => allRoles,
+    isTankJob: (job: Job) => tankJobs.includes(job),
+    isHealerJob: (job: Job) => healerJobs.includes(job),
+    isMeleeDpsJob: (job: Job) => meleeDpsJobs.includes(job),
+    isRangedDpsJob: (job: Job) => rangedDpsJobs.includes(job),
+    isCasterDpsJob: (job: Job) => casterDpsJobs.includes(job),
+    isDpsJob: (job: Job) => dpsJobs.includes(job),
+    isCraftingJob: (job: Job) => craftingJobs.includes(job),
+    isGatheringJob: (job: Job) => gatheringJobs.includes(job),
+    isCombatJob: function(job: Job) {
+      return !this.isCraftingJob(job) && !this.isGatheringJob(job);
+    },
+    canStun: (job: Job) => stunJobs.includes(job),
+    canSilence: (job: Job) => silenceJobs.includes(job),
+    canSleep: (job: Job) => sleepJobs.includes(job),
+    canCleanse: (job: Job) => cleanseJobs.includes(job),
+    canFeint: (job: Job) => feintJobs.includes(job),
+    canAddle: (job: Job) => addleJobs.includes(job),
+  };
+})();
+
+export default Util;
```

Comment:
Why IIFE?

Reply:
Mmmmm I can't remember why it was that way originally.  I'll remove the IIFE to make it cleaner.